src/storage/src/hummock/compactor/mod.rs

```diff
@@ -225,6 +227,78 @@ impl Compactor {
             .await;
         let multi_filter_key_extractor = Arc::new(multi_filter_key_extractor);
 
+        let sstable_ids = compact_task
+            .input_ssts
+            .iter()
+            .flat_map(|level| level.table_infos.iter())
+            .map(|table_info| table_info.id)
+            .collect_vec();
+
+        let compaction_size = compact_task
+            .input_ssts
+            .iter()
+            .flat_map(|level| level.table_infos.iter())
+            .map(|table_info| table_info.file_size)
+            .sum::<u64>();
+
+        let sstable_size = (context.options.sstable_size_mb as u64) << 20;
+        if compaction_size > sstable_size {
+            let mut indexes = vec![];
+            // preload the meta and get the smallest key to split sub_compaction
+            for sstable_id in sstable_ids {
+                indexes.extend(
+                    context
+                        .sstable_store
+                        .sstable(sstable_id, &mut StoreLocalStatistic::default())
+                        .await
+                        .unwrap()
+                        .value()
+                        .meta
+                        .block_metas
+                        .iter()
+                        .map(|block| {
+                            let data_size = block.len;
+                            let full_key = FullKey::from_user_key_slice(
+                                user_key(&block.smallest_key),
+                                u64::MAX,
+                            )
+                            .into_inner();
+                            (data_size, full_key.to_vec())
+                        })
+                        .collect_vec(),
+                );
+            }
+
+            // sort_by key, as for every data block has the same size;
+            indexes.sort_by(|a, b| VersionedComparator::compare_key(a.1.as_ref(), b.1.as_ref()));
+            let mut splits: Vec<KeyRange_vec> = vec![];
+            splits.push(KeyRange_vec::new(vec![], vec![]));
+            let parallelism = std::cmp::min(
+                indexes.len() as u64,
+                context.options.max_sub_compaction as u64,
+            );
+            let sub_compaction_data_size = if compaction_size > sstable_size && parallelism > 1 {
+                compaction_size / parallelism
+            } else {
+                compaction_size
+            };
+
+            if parallelism > 1 && compaction_size > sstable_size {
+                let mut last_buffer_size = 0;
+                let mut last_key: Vec<u8> = vec![];
+                for (data_size, key) in indexes {
+                    if last_buffer_size >= sub_compaction_data_size && !last_key.eq(&key) {
+                        splits.last_mut().unwrap().right = key.clone();
+                        splits.push(KeyRange_vec::new(key.clone(), vec![]));
+                        last_buffer_size = data_size as u64;
```

Comment:
move it to line 297 to unify it

Reply:
why? it is initial value, can not move to line 297